Sources/Apollo/RequestCreator.swift

```diff
@@ -124,17 +124,23 @@ extension RequestCreator {
     let operationData = try serializationFormat.serialize(value: fields)
     formData.appendPart(data: operationData, name: "operations")
 
-    var map = [String: [String]]()
-    if files.count == 1 {
-      let firstFile = files.first!
-      map["0"] = ["variables.\(firstFile.fieldName)"]
-    } else {
-      for (index, file) in files.enumerated() {
-        map["\(index)"] = ["variables.\(file.fieldName).\(index)"]
+    var map = [String: [String: String]]()
+    files.enumerated().forEach { (index, file) in
+      var field = map[file.fieldName, default: [:]]
+      if field.count == 0 {
+        field["\(index)"] = "variables.\(file.fieldName)"
+      } else if field.count == 1 {
+        field = field.mapValues { return $0 + ".0" }
+        field["\(index)"] = "variables.\(file.fieldName).\(field.count)"
+      } else {
+        field["\(index)"] = "variables.\(file.fieldName).\(field.count)"
       }
+      map[file.fieldName] = field
     }
 
-    let mapData = try serializationFormat.serialize(value: map)
+    let mapValue = Dictionary(uniqueKeysWithValues: map.flatMap { $0.1 }.map { ($0.0, [$0.1]) })
```

Comment:
This is impressively short, but it's awfully hard to tell what's going on here. Can you please give some of these variables names so it's easier to tell what's going on here?